Vite executor doesn't replace TSConfig paths for buildable libraries

### Current Behavior

Vite executor doesn't replace TSConfig paths for buildable libraries with build output paths.
This leads to `TS6059` Typescript errors (`File is not under 'rootDir'`) whenever you try to run type check on library which depends on another buildable library.

### Expected Behavior

Vite executor would replace TSConfig paths for buildable libraries with build output paths.

### GitHub Repo

https://github.com/pulecka/nx-vite-buildable-libs

### Steps to Reproduce

1. run `npm install`
2. run `npm run build` (this is alias for `nx run is-odd:build --skip-nx-cache` command)
3. see `TS6059` errors in output


### Nx Report

```shell
>  NX   Report complete - copy this into the issue template

   Node : 19.4.0
   OS   : darwin x64
   npm  : 9.2.0

   nx : 15.6.3
   @nrwl/angular : Not Found
   @nrwl/cypress : Not Found
   @nrwl/detox : Not Found
   @nrwl/devkit : 15.6.3
   @nrwl/esbuild : Not Found
   @nrwl/eslint-plugin-nx : 15.6.3
   @nrwl/expo : Not Found
   @nrwl/express : Not Found
   @nrwl/jest : Not Found
   @nrwl/js : 15.6.3
   @nrwl/linter : 15.6.3
   @nrwl/nest : Not Found
   @nrwl/next : Not Found
   @nrwl/node : Not Found
   @nrwl/nx-cloud : Not Found
   @nrwl/nx-plugin : Not Found
   @nrwl/react : Not Found
   @nrwl/react-native : Not Found
   @nrwl/rollup : Not Found
   @nrwl/schematics : Not Found
   @nrwl/storybook : Not Found
   @nrwl/web : Not Found
   @nrwl/webpack : Not Found
   @nrwl/workspace : 15.6.3
   @nrwl/vite : 15.6.3
   typescript : 4.8.4
   ---------------------------------------
   Local workspace plugins:
   ---------------------------------------
   Community plugins:
```


### Failure Logs

```shell
> nx run is-odd:build

vite v4.0.4 building for production...
✓ 4 modules transformed.
../../dist/packages/is-odd/index.mjs  0.10 kB │ gzip: 0.09 kB
../../dist/packages/is-odd/index.js  0.15 kB │ gzip: 0.15 kB

[vite:dts] Start generate declaration files...
packages/is-even/src/index.ts:1:15 - error TS6059: File '/Users/pulecka/Developer/nx-vite-buildable-libs/packages/is-even/src/lib/is-even.ts' is not under 'rootDir' '/Users/pulecka/Developer/nx-vite-buildable-libs/packages/is-odd'. 'rootDir' is expected to contain all source files.

1 export * from './lib/is-even';
                ~~~~~~~~~~~~~~~
packages/is-odd/src/lib/is-odd.ts:1:24 - error TS6059: File '/Users/pulecka/Developer/nx-vite-buildable-libs/packages/is-even/src/index.ts' is not under 'rootDir' '/Users/pulecka/Developer/nx-vite-buildable-libs/packages/is-odd'. 'rootDir' is expected to contain all source files.

1 import { isEven } from '@pulecka/is-even';
                         ~~~~~~~~~~~~~~~~~~

[vite:dts] Declaration files built in 863ms.
```


### Additional Information

_No response_

--- a/docs/generated/packages/vite/documents/set-up-vite-manually.md
+++ b/docs/generated/packages/vite/documents/set-up-vite-manually.md
@@ -83,7 +83,61 @@ You need to use the [`vite-tsconfig-paths` plugin](https://www.npmjs.com/package
 
 If you are using React, you need to use the [`@vitejs/plugin-react` plugin](https://www.npmjs.com/package/@vitejs/plugin-react).
 
+### DTS plugin
+
+If you are building a library, you need to use the [`vite-plugin-dts` plugin](https://www.npmjs.com/package/vite-plugin-dts) to generate the `.d.ts` files for your library.
+
+#### Skip diagnostics
+
+If you are building a library, you can set the `skipDiagnostics` option to `true` to speed up the build. This means that type diagnostic will be skipped during the build process. However, if there are some files with type errors which interrupt the build process, these files will not be emitted and `.d.ts` declaration files will not be generated.
+
+If you choose to skip diagnostics, here is what your `'vite-plugin-dts'` plugin setup will look like:
+
+```ts {% fileName="libs/my-lib/vite.config.ts" %}
+...
+import dts from 'vite-plugin-dts';
+import { join } from 'path';
+...
+...
+export default defineConfig({
+  plugins: [
+    ...,
+    dts({
+      entryRoot: 'src',
+      tsConfigFilePath: join(__dirname, 'tsconfig.lib.json'),
+      skipDiagnostics: true,
+    }),
+```
+
+#### Do not skip diagnostics
+
+If you are building a library, and you want to make sure that all the files are type checked, you can set the `skipDiagnostics` option to `false` to make sure that all the files are type checked. This means that type diagnostic will be run during the build process.
+
+If you choose to enable diagnostics, here is what your `'vite-plugin-dts'` plugin setup will look like:
+
+```ts {% fileName="libs/my-lib/vite.config.ts" %}
+...
+import dts from 'vite-plugin-dts';
+...
+...
+export default defineConfig({
+  plugins: [
+    ...,
+    dts({
+      root: '../../',
+      entryRoot: 'packages/one/src',
+      tsConfigFilePath: 'packages/one/tsconfig.lib.json',
+      include: ['packages/one/src/**/*.ts'],
+      outputDir: 'dist/packages/one',
+      skipDiagnostics: false,
+    }),
+```
+
+You can read more about the configuration options in the [`vite-plugin-dts` plugin documentation](https://www.npmjs.com/package/vite-plugin-dts)).
+
 ### How your `vite.config.ts` looks like
+
+#### For applications
 
 Add a `vite.config.ts` file to the root of your project. If you are not using React, you can skip adding the `react` plugin, of course.
 
@@ -102,7 +156,9 @@ export default defineConfig({
 });
 ```
 
-If you are converting a library (rather than an application) to use vite, your `vite.config.ts` file should look like this:
+#### For libraries
+
+If you are setting up a library (rather than an application) to use vite, your `vite.config.ts` file should look like this:
 
 ```ts {% fileName="libs/my-lib/vite.config.ts" %}
 import { defineConfig } from 'vite';
@@ -114,13 +170,13 @@ import { join } from 'path';
 export default defineConfig({
   plugins: [
     dts({
+      entryRoot: 'src',
       tsConfigFilePath: join(__dirname, 'tsconfig.lib.json'),
-      // Faster builds by skipping tests. Set this to false to enable type checking.
       skipDiagnostics: true,
     }),
     react(),
     viteTsConfigPaths({
-      root: '../../../',
+      root: '../../',
     }),
   ],
 

--- a/docs/shared/packages/vite/set-up-vite-manually.md
+++ b/docs/shared/packages/vite/set-up-vite-manually.md
@@ -83,7 +83,61 @@ You need to use the [`vite-tsconfig-paths` plugin](https://www.npmjs.com/package
 
 If you are using React, you need to use the [`@vitejs/plugin-react` plugin](https://www.npmjs.com/package/@vitejs/plugin-react).
 
+### DTS plugin
+
+If you are building a library, you need to use the [`vite-plugin-dts` plugin](https://www.npmjs.com/package/vite-plugin-dts) to generate the `.d.ts` files for your library.
+
+#### Skip diagnostics
+
+If you are building a library, you can set the `skipDiagnostics` option to `true` to speed up the build. This means that type diagnostic will be skipped during the build process. However, if there are some files with type errors which interrupt the build process, these files will not be emitted and `.d.ts` declaration files will not be generated.
+
+If you choose to skip diagnostics, here is what your `'vite-plugin-dts'` plugin setup will look like:
+
+```ts {% fileName="libs/my-lib/vite.config.ts" %}
+...
+import dts from 'vite-plugin-dts';
+import { join } from 'path';
+...
+...
+export default defineConfig({
+  plugins: [
+    ...,
+    dts({
+      entryRoot: 'src',
+      tsConfigFilePath: join(__dirname, 'tsconfig.lib.json'),
+      skipDiagnostics: true,
+    }),
+```
+
+#### Do not skip diagnostics
+
+If you are building a library, and you want to make sure that all the files are type checked, you can set the `skipDiagnostics` option to `false` to make sure that all the files are type checked. This means that type diagnostic will be run during the build process.
+
+If you choose to enable diagnostics, here is what your `'vite-plugin-dts'` plugin setup will look like:
+
+```ts {% fileName="libs/my-lib/vite.config.ts" %}
+...
+import dts from 'vite-plugin-dts';
+...
+...
+export default defineConfig({
+  plugins: [
+    ...,
+    dts({
+      root: '../../',
+      entryRoot: 'packages/one/src',
+      tsConfigFilePath: 'packages/one/tsconfig.lib.json',
+      include: ['packages/one/src/**/*.ts'],
+      outputDir: 'dist/packages/one',
+      skipDiagnostics: false,
+    }),
+```
+
+You can read more about the configuration options in the [`vite-plugin-dts` plugin documentation](https://www.npmjs.com/package/vite-plugin-dts)).
+
 ### How your `vite.config.ts` looks like
+
+#### For applications
 
 Add a `vite.config.ts` file to the root of your project. If you are not using React, you can skip adding the `react` plugin, of course.
 
@@ -102,7 +156,9 @@ export default defineConfig({
 });
 ```
 
-If you are converting a library (rather than an application) to use vite, your `vite.config.ts` file should look like this:
+#### For libraries
+
+If you are setting up a library (rather than an application) to use vite, your `vite.config.ts` file should look like this:
 
 ```ts {% fileName="libs/my-lib/vite.config.ts" %}
 import { defineConfig } from 'vite';
@@ -114,13 +170,13 @@ import { join } from 'path';
 export default defineConfig({
   plugins: [
     dts({
+      entryRoot: 'src',
       tsConfigFilePath: join(__dirname, 'tsconfig.lib.json'),
-      // Faster builds by skipping tests. Set this to false to enable type checking.
       skipDiagnostics: true,
     }),
     react(),
     viteTsConfigPaths({
-      root: '../../../',
+      root: '../../',
     }),
   ],
 

--- a/packages/vite/src/utils/generator-utils.ts
+++ b/packages/vite/src/utils/generator-utils.ts
@@ -491,8 +491,8 @@ export function createOrEditViteConfig(
     ? ''
     : options.includeLib
     ? `dts({
+      entryRoot: 'src',
       tsConfigFilePath: join(__dirname, 'tsconfig.lib.json'),
-      // Faster builds by skipping tests. Set this to false to enable type checking.
       skipDiagnostics: true,
     }),`
     : '';

--- a/packages/vite/src/utils/test-files/test-vite-configs.ts
+++ b/packages/vite/src/utils/test-files/test-vite-configs.ts
@@ -153,12 +153,14 @@ export const hasEverything = `
     import { defineConfig } from 'vite';
     import react from '@vitejs/plugin-react';
     import viteTsConfigPaths from 'vite-tsconfig-paths';
+    import dts from 'vite-plugin-dts';
+    import { join } from 'path';
 
     export default defineConfig({
       plugins: [
         dts({
+          entryRoot: 'src',
           tsConfigFilePath: join(__dirname, 'tsconfig.lib.json'),
-          // Faster builds by skipping tests. Set this to false to enable type checking.
           skipDiagnostics: true,
         }),
         react(),
@@ -245,8 +247,8 @@ export const testOptionObject = {
 };
 
 export const dtsPlugin = `dts({
+      entryRoot: 'src',
       tsConfigFilePath: join(__dirname, 'tsconfig.lib.json'),
-      // Faster builds by skipping tests. Set this to false to enable type checking.
       skipDiagnostics: true,
     }),`;
 export const dtsImportLine = `import dts from 'vite-plugin-dts';\nimport { join } from 'path';`;

--- a/packages/vite/src/utils/vite-config-edit-utils.spec.ts
+++ b/packages/vite/src/utils/vite-config-edit-utils.spec.ts
@@ -39,6 +39,7 @@ describe('ensureViteConfigIsCorrect', () => {
       pluginOption,
       testOption,
       testOptionObject,
+      '',
       { build: false, test: true, serve: false }
     );
     const appFileContent = tree.read('apps/my-app/vite.config.ts', 'utf-8');
@@ -63,6 +64,7 @@ describe('ensureViteConfigIsCorrect', () => {
       pluginOption,
       testOption,
       testOptionObject,
+      '',
       { build: false, test: true, serve: false }
     );
     const appFileContent = tree.read('apps/my-app/vite.config.ts', 'utf-8');
@@ -87,6 +89,7 @@ describe('ensureViteConfigIsCorrect', () => {
       pluginOption,
       testOption,
       testOptionObject,
+      '',
       { build: false, test: false, serve: false }
     );
     const appFileContent = tree.read('apps/my-app/vite.config.ts', 'utf-8');
@@ -111,6 +114,7 @@ describe('ensureViteConfigIsCorrect', () => {
       pluginOption,
       testOption,
       testOptionObject,
+      '',
       { build: false, test: false, serve: false }
     );
     const appFileContent = tree.read('apps/my-app/vite.config.ts', 'utf-8');
@@ -135,6 +139,7 @@ describe('ensureViteConfigIsCorrect', () => {
       pluginOption,
       testOption,
       testOptionObject,
+      '',
       { build: false, test: false, serve: false }
     );
     const appFileContent = tree.read('apps/my-app/vite.config.ts', 'utf-8');
@@ -159,6 +164,7 @@ describe('ensureViteConfigIsCorrect', () => {
       pluginOption,
       testOption,
       testOptionObject,
+      '',
       { build: false, test: false, serve: false }
     );
     const appFileContent = tree.read('apps/my-app/vite.config.ts', 'utf-8');
@@ -177,6 +183,7 @@ describe('ensureViteConfigIsCorrect', () => {
       pluginOption,
       testOption,
       testOptionObject,
+      '',
       { build: true, test: true, serve: true }
     );
     const appFileContent = tree.read('apps/my-app/vite.config.ts', 'utf-8');
@@ -195,6 +202,7 @@ describe('ensureViteConfigIsCorrect', () => {
       pluginOption,
       testOption,
       testOptionObject,
+      '',
       { build: false, test: true, serve: true }
     );
     const appFileContent = tree.read('apps/my-app/vite.config.ts', 'utf-8');
@@ -213,6 +221,7 @@ describe('ensureViteConfigIsCorrect', () => {
       pluginOption,
       testOption,
       testOptionObject,
+      '',
       { build: false, test: false, serve: true }
     );
     const appFileContent = tree.read('apps/my-app/vite.config.ts', 'utf-8');

